### PR TITLE
add QSLocalClient

### PR DESCRIPTION
**Pull request by HerrMuellerluedenscheid**

### CODE REVIEW

**Comment by HerrMuellerluedenscheid** [comment]
`python3 client_websocket.py` pulls the state and starts the websocket.

File: quakesaver_client/client_websocket.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        all_channels[channel] = data
+
+    return all_channels
+
+
+async def get_sensor(sensor_url):
+    async with aiohttp.ClientSession() as client:
+        async with client.get(f"http://{sensor_url}/state") as resp:
+            state = await resp.text()
+            return LocalSensor.parse_raw(state)
```
Comment (miili) [inline quakesaver_client/client_websocket.py:57]: as classmethod?

File: quakesaver_client/client_websocket.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+            async for msg in ws:
+                data = WebSocketPayload.parse_raw(msg.data)
+
+                if "data" not in data.payload:
+                    continue
+
+                waveforms = payload_to_waveforms(data)
+                logger.info(f"received {waveforms}")
+                yield waveforms
```
Comment (miili) [inline quakesaver_client/client_websocket.py:75]: as method?

Comment (HerrMuellerluedenscheid) [reply quakesaver_client/client_websocket.py]: This is now a method of the `WebsocketHandler` which adds a little more logik to manage the websocket.

File: README.md [hunk 3]
```diff
@@ -16,15 +16,15 @@ You can find the documentation [here](https://quakesaver.github.io/quakesaver-cl
...
 
 ```python
-from quakesaver_client import QSClient
+from quakesaver_client import QSCloud
 
 EMAIL = "user@yourorganisation.net"
 PASSWORD = "!verstrongpassword1"
 
-client = QSClient(email=EMAIL, password=PASSWORD)
+client = QSCloud(email=EMAIL, password=PASSWORD)
```
Comment (miili) [inline README.md:24]: `QSCloudClient`?

File: quakesaver_client/__init__.py [hunk 4]
```diff
@@ -119,3 +120,16 @@ def get_sensor(self: QSClient, sensor_uid: str) -> Sensor:
...
+    """Client to interact with sensors on your local network."""
+
+    @classmethod
+    def get_sensor(cls: QSLocal, sensor_url: str) -> LocalSensor:
+        """Get a `LocalSensor` from the url."""
+        url = f"http://{sensor_url}/state"
+        response = requests.get(url)
+        sensor = LocalSensor.parse_raw(response.text)
+        sensor.url = sensor_url
+        return sensor
```
Comment (miili) [inline quakesaver_client/__init__.py:135]: As `classmethod` of `LocalSensor`?

File: quakesaver_client/client_websocket.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        async with self.session.ws_connect(f"ws://{self.url}/ws") as ws:
+            await ws.send_str(START_ACTION.json())
+
+            async for msg in ws:
+                data = WebSocketPayload.parse_raw(msg.data)
+
+                if "data" not in data.payload:
+                    continue
+
+                trace = TraceModel(**data.payload)
```
Comment (miili) [inline quakesaver_client/client_websocket.py:65]: Unpack the `TraceModel.convert_data` straight away? (Bad name for the method)

File: quakesaver_client/client_websocket.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                convert_waveform_data(trace)
+                logger.debug(f"received data from uid: {trace.uid}")
+                yield trace
+
+    async def start(self: WebsocketHandler) -> Generator[TraceModel]:
```
Comment (miili) [inline quakesaver_client/client_websocket.py:70]: Sanity check `if not self.session`.

File: quakesaver_client/models/local_sensor.py [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+        minlatitude: float = -90,
+        maxlatitude: float = 90,
+        minlongitude: float = -180,
+        maxlongitude: float = 180,
+        level: StationDetailLevel = "station",
+        location_to_store: Path | str = None,
+    ) -> Path:
+        """Request FDSN StationXML metadata of the sensor."""
+        ...
+
```
Comment (miili) [inline quakesaver_client/models/local_sensor.py:175]: Add `async def get_waveforms(start: datetime, end: datetime, channel: str = '*')`?

File: README.md (hunk 3 above)
Comment (HerrMuellerluedenscheid) [reply README.md]: Done

File: quakesaver_client/client_websocket.py (hunk 5 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/client_websocket.py:64]: Not sure what you mean. I think it's better to have the conversion encapsulated in a dedicated function.

File: quakesaver_client/models/local_sensor.py (hunk 7 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/models/local_sensor.py]: That will be a separate MR.

File: quakesaver_client/client_websocket.py (hunk 6 above)
Comment (HerrMuellerluedenscheid) [reply quakesaver_client/client_websocket.py:74]: done. session handling refactored.

File: quakesaver_client/client_websocket.py (hunk 5 above)
Comment (miili) [reply quakesaver_client/client_websocket.py:71]: idk if it is useful to have packed binary data in the model. I would unpack the data straight away and call `.convert_data()`

Comment (HerrMuellerluedenscheid) [reply quakesaver_client/client_websocket.py:71]: `TraceModel` is autogenerated. I wouldn't want to interfere with that.

Comment (HerrMuellerluedenscheid) [reply quakesaver_client/client_websocket.py:71]: Neither would I interfere with the serialization/deserization.

**Pull request merged**